### ISSUE TITLE
Swift Package Manager 5.1.0 missing SwiftGRPC on Import

### ISSUE THREAD

**Issue by trujamal** [grpc/grpc-swift#721]
### New Issue Checklist

- [x] Updated grpc-swift to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/grpc/grpc-swift/blob/master/CONTRIBUTING.md)
- [x] I searched for [existing GitHub issues](https://github.com/grpc/grpc-swift/issues)(especially, there are many [resolved issues related to SSL](https://github.com/grpc/grpc-swift/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aclosed+ssl))

### Issue Description

Starting with an Swift UI application in Xcode

1. Single View App
2. Enter Required Info
3. Unchecked Include Unit Tests / Include UI Tests
4. Went to add package dependency
5. Used the following url `https://github.com/grpc/grpc-swift`
6. Selected Branch and typed in nio
7. Confirmed that under Package Product the only fields were GRPC | Library | Name Of Project
8. Clicked Finish
9. Went to SceneDelegate.swift then proceeded to type in import SwiftGRPC
10. Resulting error of `No such module SwiftGRPC`
11. Types in GRPC instead via this route and has no issues
12. Import the .grpc and .pb.swift files  as mentioned [here, medium starter](https://levelup.gitconnected.com/swift-grpc-577ce1a4d1b7) 
13. As a result inside of the files the following message is shown in step 10.

Unsure of how to proceed based upon the documentation in the readMe, also looking into the following [Stack Overflow thread this solution did not appear to work either](https://stackoverflow.com/questions/59488078/swift-5-specify-import-path-for-grpc-via-package-manager). Any help would be appreciated.

![alt text](https://p358.p1.n0.cdn.getcloudapp.com/items/bLumz7Dv/Image+2020-02-12+at+1.43.43+AM.png?v=0640af574c161fa7d2e7f652f785b826 "Error")

Thanks

##### Complete output when running grpc-swift, including the stack trace and command used

<details>
  <pre>No such module `SwiftGRPC`</pre>
</details>

### Environment

<details>
  <pre>
  
| Key                           | Value                                                         |
| ----------------------------- | --------------------------------------------------------------|
| OS Version                    |   10.15.3                                                       |
| Swift Version                 |   5.1.3                                                       |
| Xcode Version                 |   11.3.1                                                       |
| gRPC-Swift Version            |   1.0.0-alpha.9                                                       |
| gRPC-Swift Version            |   1.0.0-alpha.9                                                       |
| protoc Version                |   3.11.3                                                       |
| protoc-gen-swift Version      |   1.5.0                                                       |
| protoc-gen-swiftgrpc Version  |   ?.?.?                                                       |
  </pre>
</details>

**Comment by weissi** [comment]
@trujamal the module is called `GRPC`, not `SwiftGRPC`, right? Can't you just remove the `import SwiftGRPC` line and replace it by `import GRPC`?

**Comment by trujamal** [comment]
@weissi then in that case the other lines precede the code will throw an error due to the GRPC library not having the same methods. I get error messages such as unresolved identifiers for all of the methods that the protoc generates based upon the Proto file.

**Comment by demistry** [comment]
Having this same issue as well...

**Comment by demistry** [comment]
Have you been able to fix it? 
I have updated literally almost everything there is to update and still nothing..

**Comment by weissi** [comment]
CC @timburks / @MrMage

**Comment by trujamal** [comment]
@demistry The partial solution I found is based off of a separate solution thread that @nakkht left a simple answer to.

> It seems to work pretty well!
> Steps to setup:
> 
> * Sync to master branch of grpc-swift
> * Replace Package.swift contents with the following:
> 
> ```swift
> // swift-tools-version:5.1
> 
> import PackageDescription
> import Foundation
> 
> var packageDependencies: [Package.Dependency] = [
>   .package(url: "https://github.com/apple/swift-protobuf.git", .upToNextMajor(from: "1.5.0")),
>   .package(url: "https://github.com/kylef/Commander.git", .upToNextMinor(from: "0.8.0"))
> ]
> 
> var cGRPCDependencies: [Target.Dependency] = []
> 
> #if os(Linux)
> let isLinux = true
> #else
> let isLinux = false
> #endif
> 
> if isLinux || ProcessInfo.processInfo.environment.keys.contains("GRPC_USE_OPENSSL") {
>   packageDependencies.append(.package(url: "https://github.com/apple/swift-nio-ssl-support.git", from: "1.0.0"))
> } else {
>   cGRPCDependencies.append("BoringSSL")
> }
> 
> let package = Package(
>     name: "SwiftGRPC",
>     products: [
>         .library(name: "SwiftGRPC", targets: ["SwiftGRPC"])
>     ],
>     dependencies: packageDependencies,
>     targets: [
>         .target(name: "SwiftGRPC",
>                 dependencies: ["CgRPC", "SwiftProtobuf"]),
>         .target(name: "CgRPC",
>                 dependencies: cGRPCDependencies,
>                 cSettings: [
>                     .headerSearchPath("../BoringSSL/include", .when(platforms: [.iOS, .macOS, .tvOS, .watchOS])),
>                     .unsafeFlags(["-Wno-module-import-in-extern-c"])],
>                 linkerSettings: [.linkedLibrary("z")]),
>         .target(name: "protoc-gen-swiftgrpc",
>                 dependencies: [
>                     "SwiftProtobuf",
>                     "SwiftProtobufPluginLibrary",
>                     "protoc-gen-swift"]),
>         .target(name: "BoringSSL"),
>         .testTarget(name: "SwiftGRPCTests", dependencies: ["SwiftGRPC"]),
>     ],
>     swiftLanguageVersions: [.v4, .v4_2, .version("5")],
>     cLanguageStandard: .gnu11,
>     cxxLanguageStandard: .cxx11)
> ```
> 
> * Try building
> * Success!
> 
> Right after committed and referenced local grpc-swift repo with the commit to my project.
> Worked right out of the box (after removing carthage integration)
> 
> Is it something we could have with version 0.10.1?


Effectively by forking this repository (master branch / nio) into your own versioning, and updating the Package.swift file to the information above, then it will install the necessary packages. I believe the main reason why is that inside of the nio branch there are no source files or any references to SwiftGRPC unlike the master branch. Also the package.swift file inside of the nio repository may need to be updated to include the 5.1.0 tag instead of the 5.0.0 tag.

TLDR:
Fork the repository, and replace the Package.swift contents with the contents above and set the package dependency url to your fork.

**Comment by glbrntt** [comment]
@trujamal sorry for the slow response here; I've been away.

> 12. Import the .grpc and .pb.swift files  as mentioned [here, medium starter](https://levelup.gitconnected.com/swift-grpc-577ce1a4d1b7)

The tutorial linked is for the old version of (aka `SwiftGRPC`); the missing files you're seeing [here](https://github.com/grpc/grpc-swift/issues/721#issuecomment-585322211) are for that version of gRPC Swift. I.e. you're using generated code for the old version with the runtime for the new version (`GRPC`).

The instructions for generating the code from that article _broadly_ apply to the new version as well. However they are not specific enough:

> Once you have the .proto file, we need a SwiftGRPC plugin to convert the .proto files into .swift and .pb.swift files. In order to do that, we need to clone the SwiftGRPC open source project.

This is correct but **you must also checkout the version of gRPC Swift that is resolved in `Package.resolved`** (or `Package.swift` if you depend on an exact version).

> TLDR:
Fork the repository, and replace the Package.swift contents with the contents above and set the package dependency url to your fork.

This isn't necessary unless you want to build the **old** version with Xcode via Swift Package Manager.

**Comment by demistry** [comment]
On a final note use the command protoc --grpc-swift_out=. instead of protoc --swiftgrpc_out=.  to build and convert your proto files

**Comment by ohayon** [comment]
I am running into this issue currently in Swift 5.2 where it is trying to `import SwiftGRPC`, and the above fix that @demistry mentioned does not fix the issue 😞 .

**Comment by trujamal** [comment]
@ohayon use the following imports instead

<pre><code>import GRPC
import NIO
import NIOSSL
import Logging
</code></pre>

**Comment by ohayon** [comment]
Instead of `import SwiftGRPC`? And this is in the generated `.pb.swift` file right? That does not seem to fix the issue at all.

**Comment by trujamal** [comment]
> Instead of `import SwiftGRPC`? And this is in the generated `.pb.swift` file right? That does not seem to fix the issue at all.

Inside of my .pb.swift file the following imports are as follows:

<pre><code>import Foundation
import SwiftProtobuf</code></pre>

**Comment by ohayon** [comment]
Sorry, I am a bit confused here. in the generated file, these are the imports ```import Dispatch
import Foundation
import SwiftGRPC
import SwiftProtobuf```

**Comment by ohayon** [comment]
and the SwiftGRPC import fails, if i remove that, there are build failures all over for undeclared types like `ClientCallUnaryBase` and `Metadata` among many others.

**Comment by trujamal** [comment]
> Sorry, I am a bit confused here. in the generated file, these are the imports
> 
> ```
> import Dispatch
> import Foundation
> import SwiftGRPC
> import SwiftProtobuf
> ```

According to this result you are using an outdated version of the compiler located within the `local/bin/protoc-gen-grpc-swift & local/bin/protoc-gen-swift` . Last time I've updated the compiler on my end was March 22nd 2020. Based on the newly merged master branch that has the latest code.

**Comment by ohayon** [comment]
Got it! This is now working thank you! I had installed `grpc-swift` from Homebrew which appears to have not had the `protoc-gen-grpc-swift` binary in it.

**Comment by MrMage** [comment]
I guess we could either update that keg or remove it...

**Comment by ohayon** [comment]
Update would be 💯

**Comment by BrunoCerberus** [comment]
Not working
`git clone git@github.com:grpc/grpc-swift.git
cd grpc-swift
make
sudo cp protoc-gen-swift protoc-gen-grpc-swift /usr/local/bin`

![Screen Shot 2021-05-23 at 19 34 22](https://user-images.githubusercontent.com/10541956/119278799-f23aec80-bbfd-11eb-9365-944c88af7e30.png)
